src/coreclr/jit/lsraarm64.cpp

```diff
@@ -987,11 +1228,41 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         }
         else
         {
-            srcCount += BuildOperandUses(intrin.op1);
+            if ((intrin.id == NI_AdvSimd_VectorTableLookup) || (intrin.id == NI_AdvSimd_Arm64_VectorTableLookup))
+            {
+                srcCount += BuildConsecutiveRegisters(intrin.op1);
+            }
+            else
+            {
+                srcCount += BuildOperandUses(intrin.op1);
+            }
         }
     }
 
-    if ((intrin.category == HW_Category_SIMDByIndexedElement) && (genTypeSize(intrin.baseType) == 2))
+    if ((intrin.id == NI_AdvSimd_VectorTableLookup) || (intrin.id == NI_AdvSimd_Arm64_VectorTableLookup) ||
```

Comment:
I expect `VectorTableLookup` to be rarer than the others, so it might be worth putting it after `SimdByIndexedElement`

It might also be worth a flag or category to avoid needing 4-6 separate checks each time

Reply:
> It might also be worth a flag or category to avoid needing 4-6 separate checks each time

Not sure if I follow this. Can you elaborate?

Reply:
This used to be `if ((intrin.category == HW_Category_SIMDByIndexedElement) && (genTypeSize(intrin.baseType) == 2))` checked first which covers 51 different intrinsics.

Now, we're instead doing the checks against `VectorTableLookup` first and that's going to be "less common" overall.

This means the other 51 intrinsics now have to go through 4 additional branches before their path can be hit.

It would be better to handle `SIMDByIndexedElement` first and then handle `VectorTableLookup` as the `else if` block instead.

It would also be good to reduce the number of checks required to determine if this branch should be taken at all. If we mark these with a special category or flag, then we can do 1 check that covers all 4 intrinsics.